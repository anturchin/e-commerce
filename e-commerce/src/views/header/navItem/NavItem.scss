.nav-list_item {
    margin-right: 20px;
    font-size: 15px;
    font-weight: 400;
    letter-spacing: -0.01em;
    cursor: pointer;
}

.nav-list_item:hover {
    color: rgb(4, 164, 4);
}

.nav-list_item.hidden {
    display: none;
}
