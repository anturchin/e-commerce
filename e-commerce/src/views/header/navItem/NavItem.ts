import { View } from '../../View';
import './NavItem.scss';

export class NavItem extends View {
    constructor(item: string) {
        super({ tag: 'li', classNames: ['nav-list_item'], textContent: item });
    }
}
