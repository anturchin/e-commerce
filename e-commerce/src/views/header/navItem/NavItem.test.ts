import { describe, expect, test } from '@jest/globals';
import { NavItem } from './NavItem';

describe('NavItem', () => {
    test('should match', () => {
        const navItem = new NavItem('item');
        const renderedNavItem = navItem.getElement();
        expect(renderedNavItem.outerHTML).toMatchSnapshot();
    });
});
