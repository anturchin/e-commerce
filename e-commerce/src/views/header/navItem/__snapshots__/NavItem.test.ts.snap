// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem should match 1`] = `"<li class="nav-list_item" data-nav-item="item">item</li>"`;
